components/builder-scheduler/src/server/scheduler.rs

```diff
@@ -437,6 +463,58 @@ impl ScheduleMgr {
         Ok(())
     }
 
+    fn promote_package(&self, ident: &OriginPackageIdent, channel: &str) -> Result<()> {
+        debug!("Promoting '{:?}' to '{}'", ident, channel);
+
+        // We re-create the depot client instead of caching and re-using it due to the
+        // connection getting dropped when it is attempted to be re-used. This _may_ be
+        // a known issue with hyper connection pool getting reset in some cases.
+        let depot_cli = depot_client::Client::new(&self.depot_url, PRODUCT, VERSION, None).unwrap();
+
+        if channel != "stable" {
+            match depot_cli.create_channel(ident.get_origin(), channel, &self.auth_token) {
+                Ok(_) => (),
+                Err(depot_client::Error::APIError(StatusCode::Conflict, _)) => (),
+                Err(err) => {
+                    warn!("Failed to create '{}' channel: {:?}", channel, err);
+                    return Err(Error::PromoteFailure(err));
```

Comment:
Is it worth making this its own error type, e.g. `ChannelCreationFailure`, since it's technically not an error from promoting a package?